stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -1910,14 +2141,28 @@ fn delegate_stack_increase() {
         Some(pox_addr.clone()),
     );
 
+    let alice_principal = PrincipalData::from(key_to_stacks_addr(alice_key));
+    let reward_cycle = get_current_reward_cycle(&peer, &burnchain);
+    let signature =
+        make_signer_key_signature(&alice_principal, &bob_signer_private_key, reward_cycle);
+
+    println!(
+        "alice_principal: {:?}\nreward_cycle: {:?}\nsignature: {:?}\nsigning_key: {:?}",
+        &alice_principal,
+        &reward_cycle,
+        to_hex(&signature.as_slice()),
+        bob_signer_private_key.to_hex() // to_hex(&bob_signer_private_key.to)
+    );
+
     let delegate_stack_stx = make_pox_4_delegate_stack_stx(
         bob_delegate_key,
         bob_nonce,
-        PrincipalData::from(key_to_stacks_addr(alice_key)).into(),
+        alice_principal,
         min_ustx,
         pox_addr.clone(),
         block_height as u128,
         lock_period,
+        signature,
```

Comment:
Can you add some test coverage to this file for all four affected stacking functions to verify that they fail with the expected error code if the signature was bad?  Not just unit-testing the signature verification function; verifying end-to-end that the stacking function fails as expected.  Thanks!

Reply:
Added 4 new tests that cover all the stacking transactions